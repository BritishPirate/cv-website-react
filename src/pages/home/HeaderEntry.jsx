import React from "react";
import {useState} from 'react';

const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal',
  };

export default function HeaderEntry({title, link, text, svg}){ /* TODO: Create an SVG component so that I can pass it values rather than the code*/
    const [status, setStatus] = useState(STATUS.NORMAL);

    const onMouseEnter = () => {
        setStatus(STATUS.HOVERED);
    };

    const onMouseLeave = () => {
        setStatus(STATUS.NORMAL);
    };

    return(
        <a href={link}
           id={title} 
           className={'flex-item ' + status  + ' grid'} 
           onMouseEnter={onMouseEnter} 
           onMouseLeave={onMouseLeave}
           >
            <span id={title + '-icon'}>
               {svg}
            </span>
            <span id={title + '-text'} className={'text-default header-entry-text'}>{text}</span>
        </a>
    );
}
