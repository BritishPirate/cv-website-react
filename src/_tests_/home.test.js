import renderer from 'react-test-renderer';
import HeaderEntry from '../pages/home/HeaderEntry';

it('Returns a correct header entry when created', () =>{
    const component = renderer.create(
        <HeaderEntry title={'Email'} link={'mailto: nur10000@gmail.com'} text={'nur10000@gmail.com'} svg={(<svg xmlns="http://www.w3.org/2000/svg" height="24px" width="24px" fill="#FFFFFF" id="email-icon-svg">
        <path d="M0 0h24v24H0z" fill="none"/>
        <path d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"/>
        </svg>)} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
    tree.props.onMouseEnter();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});