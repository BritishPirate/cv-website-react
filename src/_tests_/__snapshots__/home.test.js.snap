// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns a correct header entry when created 1`] = `
<a
  className="flex-item normal"
  href="mailto: nur10000@gmail.com"
  id="Email"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    id="Email-icon"
  >
    <svg
      fill="#FFFFFF"
      height="24px"
      id="email-icon-svg"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>
  <span
    className="text-default"
    id="Email-text"
  >
    nur10000@gmail.com
  </span>
</a>
`;

exports[`Returns a correct header entry when created 2`] = `
<a
  className="flex-item hovered"
  href="mailto: nur10000@gmail.com"
  id="Email"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    id="Email-icon"
  >
    <svg
      fill="#FFFFFF"
      height="24px"
      id="email-icon-svg"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>
  <span
    className="text-default"
    id="Email-text"
  >
    nur10000@gmail.com
  </span>
</a>
`;

exports[`Returns a correct header entry when created 3`] = `
<a
  className="flex-item normal"
  href="mailto: nur10000@gmail.com"
  id="Email"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    id="Email-icon"
  >
    <svg
      fill="#FFFFFF"
      height="24px"
      id="email-icon-svg"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
  </span>
  <span
    className="text-default"
    id="Email-text"
  >
    nur10000@gmail.com
  </span>
</a>
`;
